validation-test/stdlib/Collection/LazyFilterCollection.swift

```diff
@@ -0,0 +1,146 @@
+// -*- swift -*-
+
+// RUN: rm -rf %t
+// RUN: mkdir -p %t
+// RUN: %S/../../../utils/gyb %s -o %t/main.swift
+// RUN: %S/../../../utils/line-directive %t/main.swift -- %target-build-swift %t/main.swift -o %t/LazyFilterCollection.swift.a.out
+// RUN: %S/../../../utils/line-directive %t/main.swift -- %target-run %t/LazyFilterCollection.swift.a.out
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import StdlibCollectionUnittest
+
+// Also import modules which are used by StdlibUnittest internally. This
+// workaround is needed to link all required libraries in case we compile
+// StdlibUnittest with -sil-serialize-all.
+import SwiftPrivate
+#if _runtime(_ObjC)
+import ObjectiveC
+#endif
+
+var CollectionTests = TestSuite("Collection")
+
+// Test collection as sequence using value types as elements.
+CollectionTests.addBidirectionalCollectionTests(
+  makeCollection: { (elements: [OpaqueValue<Int>]) in
+    MinimalSequence(elements: elements).lazy.filter { _ in return true }
+  },
+  wrapValue: identity,
+  extractValue: identity,
+  makeCollectionOfEquatable: { (elements: [MinimalEquatableValue]) in
+    MinimalSequence(elements: elements).lazy.filter { _ in return true }
```

Comment:
I'm not sure if Jira was the best place to file a ticket, given that this is not part of master. LMK if you want me to delete the ticket. Otherwise, a description and sample code can be found there.